src/apps/checklist-button/checklist-material-button.slice.jsx

```diff
@@ -40,31 +53,44 @@ export const checklistMaterialButtonSlice = createSlice({
   name: "checklistMaterial",
   initialState: { status: {}, onList: {} },
   reducers: {
+    setInitialStatus(state, action) {
+      state.status[action.payload.materialId] = "ready";
+      state.onList[action.payload.materialId] = action.payload.onList;
+    },
     addToListPending(state, action) {
       state.status[action.payload.materialId] = "pending";
-      state.onList[action.payload.materialId] = "on";
+      state.onList[action.payload.materialId] = "off";
     },
     addToListAborted(state, action) {
       state.status[action.payload.materialId] = "failed";
       state.onList[action.payload.materialId] = "off";
     },
     removeFromListPending(state, action) {
       state.status[action.payload.materialId] = "pending";
-      state.onList[action.payload.materialId] = "off";
+      state.onList[action.payload.materialId] = "on";
     },
     removeFromListAborted(state, action) {
       state.status[action.payload.materialId] = "failed";
       state.onList[action.payload.materialId] = "on";
     }
   },
   extraReducers: {
+    [checkOnListAction.pending]: (state, action) => {
+      state.onList[action.meta.arg.materialId] = "unknown";
+    },
+    [checkOnListAction.fulfilled]: (state, action) => {
+      state.onList[action.meta.arg.materialId] = action.payload ? "on" : "off";
```

Comment:
What will the payload of the `checkOnListAction` be if the check fails and `rejectWithValue(err)` is returned?